kazoo/recipe/lock.py

```diff
@@ -267,6 +303,56 @@ def __exit__(self, exc_type, exc_value, traceback):
         self.release()
 
 
+def WLock(*args, **kwargs):
```

Comment:
Why aren't these subclasses as is typical in Python to slightly modify behavior? The lack of subclassing here means that when you look at tracebacks and such, there's nothing indicating WLock vs RLock vs Lock. That will drive people crazy.